app/services/nomis/elite2/prison_offender_manager.rb

```diff
@@ -29,6 +29,31 @@ class PrisonOffenderManager
       def full_name
         "#{last_name}, #{first_name}".titleize
       end
+
+      def add_detail(pom_detail)
+        allocations = pom_detail.allocations.where(active: true)
+        allocation_counts = get_allocation_counts(allocations)
+        self.tier_a = allocation_counts['A']
+        self.tier_b = allocation_counts['B']
+        self.tier_c = allocation_counts['C']
+        self.tier_d = allocation_counts['D']
+        self.total_cases = [tier_a, tier_b, tier_c, tier_d].sum
+        self.status = pom_detail.status
+        self.working_pattern = pom_detail.working_pattern
+      end
+
+    private
+
+      def get_allocation_counts(allocations)
```

Comment:
https://makandracards.com/makandra/31141-ruby-counting-occurrences-of-an-item-in-an-array-enumerable